common/src/program_storage.rs

```diff
@@ -79,34 +83,41 @@ pub trait ProgramStorage {
     fn update_active_program<F, ReturnType>(
         program_id: ProgramId,
         update_action: F,
+        block_number: u32,
     ) -> Result<ReturnType, Self::Error>
     where
         F: FnOnce(&mut ActiveProgram) -> ReturnType,
     {
-        Self::update_program_if_active(program_id, |program| match program {
-            Program::Active(active_program) => update_action(active_program),
-            _ => unreachable!("invariant kept by update_program_if_active"),
-        })
+        Self::update_program_if_active(
+            program_id,
+            |program| match program {
+                Program::Active(active_program) => update_action(active_program),
+                _ => unreachable!("invariant kept by update_program_if_active"),
+            },
+            block_number,
+        )
     }
 
     /// Update the program under the given key `program_id` only if the
     /// stored program is an active one.
     fn update_program_if_active<F, ReturnType>(
         program_id: ProgramId,
         update_action: F,
+        block_number: u32,
```

Comment:
let's mutate bn with closure instead of extra argument

Reply:
done